client/data/README.md

```diff
@@ -0,0 +1,33 @@
+# Data
+
+This directory contains all services related to data fetching. Folders within this directory refer to domains of the application that are exposed through the services.
+
+Behind the abstraction, they are using `react-query` as the way to query and mutate data from the back-end. This is inline with the new approach on data fetching that Calypso is taking.
```

Comment:
I'd generally advise against using the word "new" in that kind of documentation. Just because it can get obsolete quickly, and to no longer be "new", and that can get confusing if we forget to update it.